sbc_compassion/migrations/12.0.1.0.4/post-migration.py

```diff
@@ -0,0 +1,48 @@
+from openupgradelib import openupgrade
+
+
+def _update_reading_language(cr, sponsorship_id, lang):
+    cr.execute(
+        "UPDATE recurring_contract "
+        "SET reading_language = %s "
+        "WHERE id = %s" % (lang, sponsorship_id)
+    )
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    if not version:
+        return
+
+    sponsorships_to_fill = env["recurring.contract"].search([
+        ("reading_language", "=", False)
```

Comment:
We can limit to child sponsorships that are not yet finished.
```suggestion
        ("reading_language", "=", False),
        ("child_id", "!=", False),
        ("state", "not in", ["terminated", "cancelled"])
```